docs/OV_Runtime_UG/ShapeInference.md

```diff
@@ -92,16 +94,16 @@ Shape collision during shape propagation may be a sign that a new shape does not
 Changing the model input shape may result in intermediate operations shape collision.
 
 Examples of such operations:
-* [Reshape](../ops/shape/Reshape_1.md) operation with a hard-coded output shape value
-* [MatMul](../ops/matrix/MatMul_1.md) operation with the `Const` second input cannot be resized by spatial dimensions due to operation semantics
+* [Reshape](../ops/shape/Reshape_1.md) operation with a hard-coded output shape value.
+* [MatMul](../ops/matrix/MatMul_1.md) operation with the `Const` second input cannot be resized by spatial dimensions due to operation semantics.
```

Comment:
> 

Hmm, I'd leave it the way it was. An input "not being able to be ....." sounds too perplexing, IMHO. 
I also think such phrases would go better when referring to animate objects.

Reply:
That has changed the meaning: the "cannot" version means "MatMul operation cannot be resized"; the "not being able to" version means "the `Const` second input cannot be resized". I think an operation cannot be resized, only the input can, right?

Reply:
What about:

```suggestion
* [MatMul](../ops/matrix/MatMul_1.md) operation with the `Const` second input that cannot be resized by spatial dimensions due to operation semantics.
```

?? ;)

Reply:
That can still be ambiguous because the object of the clause can still be "MatMul". Or: [MatMul](../ops/matrix/MatMul_1.md) operation with the `Const` second input, where the `Const` second input cannot resized by spatial dimensions due to operation semantics.
Not sure if it will make the sentence even more complicated. 
The "with sth doing" phrase is more clear to me.

Reply:
Speaking of ambiguity, I believe same could be applied to:

```
an operation with the input not being able to be resized
```
The safest approach will be to split it into two sentences.

```suggestion
* [MatMul](../ops/matrix/MatMul_1.md) operation with the `Const` second input. The input cannot be resized by spatial dimensions due to operation semantics.
```

Reply:
I think "an operation with the input not being able to be resized" does not have this ambiguity, because the phrase "not being able to be resized" is not a standard attributive clause. It only makes sense with a "with", which makes "the input" the object. 
If you want to write it into two sentences, your need to be clear on one question: can the `Const` second input be resized alone? or only when it cannot be resized applies to the case here?

Reply:
Being clear on this:

```suggestion
* [MatMul](../ops/matrix/MatMul_1.md) operation with the `Const` second input. This input cannot be resized by spatial dimensions due to operation semantics.
```
Thus, we are certain it refers to the input that is mentioned.

Reply:
We can integrate it by adding an "and".

Reply:
I think so too.

```suggestion
* [MatMul](../ops/matrix/MatMul_1.md) operation with the `Const` second input and this input cannot be resized by spatial dimensions due to operation semantics.
```